p2p/net/conn.go

```diff
@@ -229,9 +261,13 @@ func (c *FormattedConnection) closeUnlocked() error {
 // Close closes the connection (implements io.Closer). It is go safe.
 func (c *FormattedConnection) Close() error {
 	c.wmtx.Lock()
+	defer c.wmtx.Unlock()
 	err := c.closeUnlocked()
```

Comment:
Does calling closeUnlocked more than once causes an error for sure? if not then you might try closing c.stopSending more than once.